contentcuration/contentcuration/frontend/shared/views/channel/ChannelModal.vue

```diff
@@ -310,17 +312,21 @@
         });
       },
       close() {
-        this.$router.push({
-          name: this.$route.query.last,
-          params: this.$route.params,
-          query: {
-            // we can navigate to this component
-            // from the catalog search page =>
-            // do not lose search query
-            ...this.$route.query,
-            last: undefined,
-          },
-        });
+        if (this.$route.query.last) {
+          this.$router.push({
+            name: this.$route.query.last,
+            params: this.$route.params,
+            query: {
+              // we can navigate to this component
+              // from the catalog search page =>
+              // do not lose search query
```

Comment:
Is coming here from search just a "for example" or something that actually could/would happen? It may be worth clarifying it more generally either way so it's a bit clearer that - in any case - coming to this page from a route that had query params will be rerouted with those params.

Reply:
That's a really good question.  This comment is left from the original implementation, and I couldn't tell to what degree we depend on this behavior.  At this point it seems hypothetical, since the catalog view only links to the channel details modal, even for channels that you have edit access to.  But I could see that changing at some point.

@rtibbles do you have any thoughts on this?

Reply:
@micahscopes That's right. I think I might have copied that from `ChannelDetailsModal` and forgot to check if we really navigate to this one from the catalog. It seems that this route query logic should be removed here indeed as it can be misleading (though we still need to keep `last` and `params` because we navigate to it from at least two pages).